examples/wasip3-streaming/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+use spin_sdk::http_wasip3::body::IncomingBodyExt;
+use spin_sdk::http_wasip3::http_service;
+use spin_sdk::http_wasip3::{IntoResponse, Request};
+
+use bytes::Bytes;
+use futures::{
+    channel::mpsc::{channel, Sender},
+    SinkExt, StreamExt,
+};
+
+/// A streaming Spin HTTP component.
+///
+/// This component is an (almost) pure echo server: it streams the
+/// request body back to the client, with a start and end marker
+/// for UI visibility. A real service would of course process the
+/// request in some way!
+#[http_service]
+async fn handle_wasip3_streaming(request: Request) -> impl IntoResponse {
+    // We want to process the incoming body as a stream. This allows
+    // us to send the first bytes of the response without waiting
+    // for the request to arrive in full (and without requiring us to
+    // hold the whole request body in memory).
+    let mut in_body = request.into_body().stream();
+
+    // Create a streaming Body implementation that backs onto a `mpsc`
+    // channel. The function returns the sender side of the channel; the
+    // receiver end becomes the body. So anything written to the sender
+    // side will be sent out over the HTTP response.
+    let (mut tx, body) = bytes_stream_body();
+
+    // Use wit_bindgen::spawn to allow the async block to keep running
+    // after the handler returns.
+    spin_sdk::http_wasip3::wasip3::wit_bindgen::spawn(async move {
```

Comment:
Should we be exposing `spawn` from the SDK directly instead of requiring users to know the very long `spin_sdk::http_wasip3::wasip3::wit_bindgen` namespace?

Reply:
Excellent question - we may want to consider that for `wit_future` and `wit_stream` as well.  I'd like to put that out of scope for this PR though if that's okay?